### PR TITLE
Fix state prefixes, minor refactor to reduce API surface

### PR DESCRIPTION
**Pull request by ValarDragon**

### CODE REVIEW

File: x/gamm/keeper/swap.go [hunk 1]
```diff
@@ -225,11 +225,11 @@ func (k Keeper) CalculateSpotPrice(ctx sdk.Context, poolId uint64, tokenInDenom,
 		inRecord.Weight.ToDec(),
 		outRecord.Token.Amount.ToDec(),
 		outRecord.Weight.ToDec(),
-		poolAcc.GetPoolParams().SwapFee,
+		sdk.ZeroDec(),
 	), nil
 }
 
-func (k Keeper) CalculateSpotPriceSansSwapFee(ctx sdk.Context, poolId uint64, tokenInDenom, tokenOutDenom string) (sdk.Dec, error) {
+func (k Keeper) CalculateSpotPriceWithSwapFee(ctx sdk.Context, poolId uint64, tokenInDenom, tokenOutDenom string) (sdk.Dec, error) {
```
Comment (ValarDragon) [inline x/gamm/keeper/swap.go:232]: woops didn't mean to commit this file

File: x/gamm/keeper/pool.go [hunk 2]
```diff
@@ -50,7 +52,8 @@ func (k Keeper) NewPool(ctx sdk.Context, poolId uint64, poolParams types.PoolPar
 	return poolAcc, nil
 }
 
-func (k Keeper) GetNextPoolNumber(ctx sdk.Context) uint64 {
+// getNextPoolNumber returns the next pool number
+func (k Keeper) getNextPoolNumber(ctx sdk.Context) uint64 {
```
Comment (sunnya97) [inline x/gamm/keeper/pool.go:56]: Why do you want this to be private?

Comment (ValarDragon) [reply x/gamm/keeper/pool.go:56]: Wanted to reduce the API surface of the keeper

Comment (sunnya97) [reply x/gamm/keeper/pool.go:56]: Oh I didn't realize it also incremented the value, I thought it just peeked.  Yeah, then should make it private.

File: x/gamm/types/account_test.go [hunk 3]
```diff
@@ -381,7 +381,7 @@ func TestPoolAccountMarshalYAML(t *testing.T) {
 	require.NoError(t, err)
 
 	want := `|
-  address: cosmos10vgzkwmgvqs5nwh3mprrt53ff6r7autptppdxh
+  address: cosmos1m48tfmd0e6yqgfhraxl9ddt7lygpsnsrhtwpas
```
Comment (sunnya97) [inline x/gamm/types/account_test.go:384]: What is this for?

Comment (ValarDragon) [reply x/gamm/types/account_test.go:384]: This is the module address, but it has changed due to the store keys changing

**Review by sunnya97 — APPROVED** [review]

**Pull request merged**